cpp/src/parquet/column_reader_test.cc

```diff
@@ -317,13 +318,64 @@ TEST_F(TestPrimitiveReader, TestInt32FlatRequiredSkip) {
       values_.end());
   ASSERT_TRUE(vector_equal(sub_values, vresult));
 
+  // 4) skip_size = 0
+  levels_skipped = reader->Skip(0);
+  ASSERT_EQ(0, levels_skipped);
+
+  // 5) Skip past the end page. There are 5 pages and we have either skipped
+  // or read all of them, so there is nothing left to skip.
+  levels_skipped = reader->Skip(10);
+  ASSERT_EQ(0, levels_skipped);
+
   values_.clear();
   def_levels_.clear();
   rep_levels_.clear();
   pages_.clear();
   reader_.reset();
 }
 
+// Skip with repeated field. This test makes it clear that we are skipping
+// values and not records.
+TEST_F(TestPrimitiveReader, TestSkipRepeatedField) {
+  // Example schema: message M { repeated int32 b = 1 }
+  max_def_level_ = 1;
+  max_rep_level_ = 1;
+  NodePtr type = schema::Int32("b", Repetition::REPEATED);
+  const ColumnDescriptor descr(type, max_def_level_, max_rep_level_);
+  // Example rows: {}, {[10, 10]}, {[20, 20, 20]}
+  std::vector<int32_t> values = {10, 10, 20, 20, 20};
+  std::vector<int16_t> def_levels{0, 1, 1, 1, 1, 1};
+  std::vector<int16_t> rep_levels = {0, 0, 1, 0, 1, 1};
+  std::shared_ptr<DataPageV1> page =
+      MakeDataPage<Int32Type>(&descr, values, /*num_values=*/def_levels.size(),
+                              Encoding::PLAIN, /*indices=*/{},
+                              /*indices_size=*/0, def_levels, max_def_level_,
+                              rep_levels, max_rep_level_);
+
+  pages_.push_back(std::move(page));
+
+  InitReader(&descr);
+  Int32Reader* reader = static_cast<Int32Reader*>(reader_.get());
+
+  std::vector<int32_t> vresult(4, -1);
```

Comment:
nit: could we make make variable names slightly more descriptive?

Reply:
Addressed the comments, PTAL.